### PR TITLE
Add method to set custom hostname

### PR DESCRIPTION
**Pull request by microDev1**
Running the following code will set custom hostname of wifi interface.

```python
import wifi
wifi.radio.set_hostname("custom-hostname")
```

### CODE REVIEW

**Comment by askpatrickw** [comment]
@microDev1  Looks like you're using the Config instead of falling back to the USB Device name. Do I have that correct? Will https://github.com/adafruit/circuitpython/pull/3530 still be needed?

File: shared-bindings/wifi/Radio.c [hunk 1]
```diff
@@ -102,6 +102,26 @@ STATIC mp_obj_t wifi_radio_stop_scanning_networks(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_stop_scanning_networks_obj, wifi_radio_stop_scanning_networks);
 
+//|     def set_hostname(self, hostname: ReadableBuffer) -> None:
```
Comment (tannewt) [inline shared-bindings/wifi/Radio.c:105]: Please do this as a property instead. That way it's simpler to read and write it.

**Review by tannewt — CHANGES_REQUESTED** [review]

**Comment by microDev1** [comment]
@askpatrickw You are correct... with the current implementation #3530 is still needed.

File: shared-bindings/wifi/Radio.c [hunk 2]
```diff
@@ -102,6 +101,38 @@ STATIC mp_obj_t wifi_radio_stop_scanning_networks(mp_obj_t self_in) {
...
+    return common_hal_wifi_radio_get_hostname(self);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_get_hostname_obj, wifi_radio_get_hostname);
+
+STATIC mp_obj_t wifi_radio_set_hostname(mp_obj_t self_in, mp_obj_t hostname_in) {
+    mp_buffer_info_t hostname;
+    mp_get_buffer_raise(hostname_in, &hostname, MP_BUFFER_READ);
+
+    if (hostname.len < 1 || hostname.len > 63) {
+        mp_raise_ValueError(translate("Hostname must be between 1 and 63 characters"));
```
Comment (tannewt) [inline shared-bindings/wifi/Radio.c:119]: Can this handle unicode? Maybe we should take a string explicitly.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question about unicode.

File: shared-bindings/wifi/Radio.c (hunk 2 above)
Comment (dhalbert) [reply shared-bindings/wifi/Radio.c:119]: DNS hostnames can only be English language letters, digits, and hyphens. I don't know if the ESP32 firmware checks for that.

Comment (tannewt) [reply shared-bindings/wifi/Radio.c:119]: It doesn't look like it's validated: https://github.com/espressif/esp-idf/blob/8bc19ba893e5544d571a753d82b44a84799b94b1/components/esp_netif/lwip/esp_netif_lwip.c#L1108

**Review by tannewt — APPROVED** [review]
Awesome! Thank you!

**Pull request merged**